synapseclient/client.py

```diff
@@ -2039,7 +2041,24 @@ def getTeamMembers(self, team):
         for result in self._GET_paginated('/teamMembers/{id}'.format(id=id_of(team))):
             yield TeamMember(**result)
 
-    def submit(self, evaluation, entity, name=None, team=None, silent=False, submitterAlias=None, teamName=None):
+    def _get_docker_digest(self, entity, docker_tag):
+        '''
+        Get matching Docker sha-digest of a DockerRepository given a Docker tag
+        
+        :param entity:      Synapse id of Docker repository
+        :param docker_tag:  Docker tag
+        :returns: Docker digest matching Docker tag
+        '''
+        docker_tags = self.restGET('/entity/{entityId}/dockerTag'.format(entityId=entity.id)) 
+        docker_digest = None
+        for tag in docker_tags['results']:
+            if docker_tag == tag['tag']:
+                docker_digest = tag['digest']
+        if docker_digest is None:
+            raise ValueError("Docker tag {docker_tag} not found.  Please specify a docker tag that exists. The dockerTag 'latest' is used as default.".format(docker_tag=docker_tag))
+        return(docker_digest)
+
+    def submit(self, evaluation, entity, name=None, dockerTag="latest", team=None, silent=False, submitterAlias=None, teamName=None):
```

Comment:
It's dangerous to add a parameter in the middle of the method signature. Old code that depends on the key word argument to appear at the right position will break.